### ISSUE TITLE
Global configuration for JVM for bloop

### ISSUE THREAD

**Issue by romanowski** [VirtusLab/scala-cli#1386]
**Is your feature request related to a problem? Please describe.**

Scala CLI wants to stick to default, LTS JDK - for now it is Java 17. Some users wants to use newer versions like 18 or 19 and Scala CLI should have a easy way to configure that.

**Describe the solution you'd like**

A global setting that specify default JDK for bloop. It should accept same arguments as `--jvm`. Also it should not allow JDK lower then 17. 

Additionally, we should update the error, warning and info messages about JDK that we report.

**Additional context**

Inspired by @milessabin  [comment](https://github.com/VirtusLab/scala-cli/issues/1379#issuecomment-1252506791).

**Comment by SethTisue** [comment]
@MaciejG604 and I are looking at this at today's Tooling Spree

>  It should accept same arguments as --jvm

It's not clear to me if that's the right design, or if the right design is to be like `--java-home`. What if the JVM the user wants to use is somewhere where Coursier doesn't know about?

**Comment by SethTisue** [comment]
Note that `--bloop-jvm` does already exist. I hadn't realized that until just now.

But what we don't have is `--bloop-java-home`, to cover the case where Coursier doesn't know about the JVM the user wants to use.

So the main task in front of us is:

* add `bloop-jvm` config setting (corresponding to `--bloop-jvm`)

but we should also consider

* add `--bloop-java-home`
* add `bloop-java-home` config setting

**Comment by SethTisue** [comment]
We have apparently-working WIP code for the config setting, but there's a strange thing where `scala-cli bloop` and `scala-cli bloop start` seem to have independent implementations (in `Bloop` and `BloopStart`). What we have works for `bloop`, do we just copy-and-paste the same code into `BloopStart`? Seems strange that a duplicate would be required

We also need to do some error checking and handling for whether the requested Java version is 17+. Do we do that at the time the user sets the config, or check it at the time Bloop starts, or both?

We need to write a test. For now, we are testing it manually with:

```
./mill -i scala config bloop-jvm zulu:17
scala-cli --power bloop exit
./mill -i scala --power bloop
./mill -i scala --power bloop about
```

which prints a message such as:

```
Running on Java JDK v17.0.14 (/Users/tisue/Library/Caches/Coursier/arc/https/cdn.azul.com/zulu/bin/zulu17.56.15-ca-jdk17.0.14-macosx_aarch64.tar.gz/zulu17.56.15-ca-jdk17.0.14-macosx_aarch64/zulu-17.jdk/Contents/Home)
```

**Comment by SethTisue** [comment]
wip code: https://github.com/SethTisue/scala-cli/commits/issue-1386/

**Comment by Gedochao** [comment]
> What we have works for bloop, do we just copy-and-paste the same code into BloopStart?

Yeah.... that's awkward, but likely what we need to do. Some abstraction might be a good idea (just leave a `TODO` comment, if you don't want to refactor). 

> We also need to do some error checking and handling for whether the requested Java version is 17+. Do we do that at the time the user sets the config, or check it at the time Bloop starts, or both?

Both.
